src/mpyl/utilities/docker/__init__.py

```diff
@@ -338,3 +350,72 @@ def remove_container(logger: Logger, container: Container) -> None:
     logger.debug(f"Removing container {container.id}")
     docker.remove(container.id)
     logger.info(f"Removed container {container.id}")
+
+
+def create_ecr_repo_if_needed(
+    logger: Logger, registry_config: DockerRegistryConfig, repo: str
+):
+    ecr_config = Config(
+        region_name=registry_config.region,
+        signature_version="v4",
+        retries={"max_attempts": 10, "mode": "standard"},
+    )
+    ecr_client = boto3.client(
+        "ecr",
+        config=ecr_config,
+        aws_access_key_id=registry_config.user_name,
+        aws_secret_access_key=registry_config.password,
+    )
+    try:
+        ecr_client.describe_repositories(
+            repositoryNames=[
+                repo.lower(),
+            ]
+        )
+        logger.info(f"Repository '{repo}' exists.")
+    except (
+        ecr_client.exceptions.InvalidParameterException,
+        ecr_client.exceptions.ServerException,
+    ) as exc:
+        raise exc
```

Comment:
You can remove the excepts (also on line 396) in which you just raise the exception anyway right? That would happen anyway without the except, or am I missing something?

Reply:
Yeah I had a comment about the exceptions but I didn't mean to be listed all like this. Was just checking that you put some ecr method calls in a `try ... except` block, but then had other ecr methods in the except block as well (those will not be caught if they throw exceptions). Is that what you had in mind?

Reply:
I'm unsure of any other exceptions that could be thrown.

Reply:
added a lifecycle policy check exception